0.24

balloob opened the pull request:
**Read the breaking changes on how to migrate your history**

It's time for 0.24 and it contains some structural changes to our history which requires action from your end, so please keep reading.

Roy Hooper did an amazing job migrating the history support from being tied to SQLite to use the ORM SQLAlchemy instead. This means that you can now use **any** SQL backend for the history. So besides SQLite you can now databases like MySQL or PostgreSQL.

However, this does require that you run a command to migrate your existing history over. We tried to make the process as seamless as possible by introducing a new command line script: 

``` bash
hass --script db_migrator --config /path/to/config
```

You can omit the `--config` option if you use the default configuration directory. Run the script with `--help` to get more options.

Other changes:
- Doing an epic job making sure all requirements and docs are up to date (@fabaff)
- Alarm: SimpliSafe is now supported (@w1ll1am23)
- Add scripts to command line to expose advanced options (@balloob)
- Media Player: Plex will no longer spam the logs if server goes offline (@dale3h)
- Sensor: APCUPSd now supports names, icons and units (@dale3h)
- Lock: Verisure entities will now use name instead of serial number for entity id (@turbokongen)
- Statsd can now also export attributes (@bah2830)
- Sensor: Yahoo! Weather now suppoted (@pvizeli)
- Switch: TPLink HS100/HS110 now supported (@GadgetReactor)
- Support for KNX added (@usul27)
- Stability fixes for RFXTRX (@Danielhiversen)
- Tweaks to Z-Wave (@turbokongen)
- Light: brightness now clamped to 0-255 (@keatontaylor)
- Support for Join by joaoapps (@nkgilley)
- Thermostat: Radiotherm HVAC mode now supported (@danieljkemp)
- Google Travel times can now use dynamic locations for start/end (@bah2830)
- Notify: Allow sending photos to Telegram (@pvizeli)
- Frontend: Improve loading times (@balloob)
- Frontend: fix stream not reconnecting after standby (@balloob)
- Frontend: Wait up to two seconds for new state before resetting toggle after toggling state (@balloob)
### Breaking changes
- Entity ids of Verisure locks will change. This is a one time change but should improve readability.
- APCUPSd settings have changed to accommodate the extended options

balloob committed:
Version bump to 0.24.0.dev0

fabaff committed:
Docstring updates (#2404)

* Fix docstring

* Fix typo

* Update docstrings

* Update docstrings

fabaff committed:
Move content to devel docs (fixes #2403) (#2408)

rhooper committed:
bump pyvera version to 0.2.13 (#2406)

fabaff committed:
Fix update (#2402)

fabaff committed:
Remove unused links (#2417)

emlove committed:
Move Aeotec bulb color logic to Zwave workaround

Default behavior for warm/cold white channels is to assume the white
channel is mixed with the rgb. This is a sane default and should support
the Fibaro RGBW LED controller.

chennin committed:
check for OP_NO_COMPRESSION support before trying to use it (#2423)

Commit:
Downgraded pubnub version (#2420)

fabaff committed:
A mini update (#2418)

balloob committed:
Merge pull request #2421 from armills/zwave-color-bulbs

Move Aeotec bulb color logic to Zwave workaround

fabaff committed:
Upgrade fuzzywuzzy to 0.11.0 (#2416)

fabaff committed:
Upgrade slacker to 0.9.18 (#2415)

fabaff committed:
Upgrade python-telegram-bot to 4.3.1 (#2414)

rhooper committed:
support newer deCONZ api versions (#2410)

Commit:
SimpliSafe Alarm (#2409)

fabaff committed:
Use XML source instead of website (#2400)

rhooper committed:
Switch to SQLAlchemy for the Recorder component.  Gives the ability t… (#2377)

* Switch to SQLAlchemy for the Recorder component.  Gives the ability to use MySQL or other.

* fixes for failed lint

* add conversion script

* code review fixes and refactor to use to_native() model methods and execute() helper

* move script to homeassistant.scripts module

* style fixes my tox lint/flake8 missed

* move exclusion up

balloob committed:
Make scripts available via CLI (#2426)

* Rename sqlalchemy migrate script

* Add script support to CLI

Commit:
Updated to version 3.1.2 and fixed invalid host setup error (#2431)

Commit:
Updated to new statsd library and added state change counters (#2429)

balloob committed:
properly cleanup after config test

kellerza committed:
Persistent_notification service description (#2407)

* Persistent_notification service description

* Add service name to services.yaml

Commit:
Update unifi.py to support sites (#2434)

* Update unifi.py

Add support for a site that is not the default within the Unifi Controller.

i.e. A controller with multiple sites:

 - Home
 - Friends
 - Parents (default)

Supplying the identifier for 'Home' now means that the devices tracked will be associated with 'Home'.

* Update test_unifi.py

Fix test modules as well.

balloob committed:
Update frontend

fabaff committed:
Change schema for elevation to int (#2436)

dale3h committed:
Fix flood of errors if Plex server goes offline (#2447)

fabaff committed:
Upgrade python-telegram-bot to 4.3.2 (#2459)

fabaff committed:
Upgrade slacker to 0.9.21 (#2458)

dale3h committed:
Add names, units and icons to APCUPSd Sensor (#2443)

* Add names, units and icons to APCUPSd Sensor

* Fix farcy errors

* Attempt fix of errors

* Remove "type:" from configuration

* Remove duplicate "mdi:" prefix

fabaff committed:
Upgrade pyowm to 2.3.2 (fixes #2452) (#2464)

turbokongen committed:
Return name of location to lock instead of serial number. (#2460)

tchellomello committed:
- Added code validation on Simplisafe module on alarm_control_panel (#2455)

kellerza committed:
Exclude secrets.yaml in yaml !include_directories (#2450)

laf committed:
Updated braviatv media player to support power status (#2470)

* Updated braviatv media player to support power status

* Updated requirements_all.txt

balloob committed:
Fix PyLint 1.6 issues (#2471)

Commit:
Added attributes to the statsd data (#2440)

* Added attributes to the statsd data

* Updated to allow optional attribute logging

turbokongen committed:
Fix missing generic command class for binary sensors (#2475)

pvizeli committed:
Yahoo! weather support (#2457)

* initial import yahoo weather

* fix temperature in HA style

* add suggestion from @fabaff

* change with suggestion from @balloob

dale3h committed:
Fix AsusWRT to prevent SSH key confusion (#2467)

Changed "pub_key" to "ssh_key" while maintaining backwards compatibility. Quotes were also updated to match across the file.

koen01 committed:
Add 'Sound' to rfxtrx DATA_TYPES (#2477)

Fixes reception of SelectPlus and correctly adds the chime sensor.

clach04 committed:
Fix typos in demo switch doc strings (#2480)

GadgetReactor committed:
New Switch Platform: TPLink Switch (HS100 / HS110) (#2453)

* New Switch Platform: TPLink Switch (HS100 / HS110)

### Information

The TPLink switch platform allows you to control the state of your TPLink Wi-Fi Smart Plugs.

Supported devices (tested):
HS100 (UK)

It should also work with the HS110.

To use your D-Link smart plugs in your installation, add the following to your configuration.yaml file:

"""
# Example configuration.yaml entry
switch:
  platform: tplink
  host: IP_ADRRESS
  name: TPLink Switch
"""

### Configuration variables:

host (Required): The IP address of your TPlink plug, eg. http://192.168.1.105
name (Optional): The name to use when displaying this switch.

* Update tplink.py

Bug fixes

* Separate to a standalone library

* Removed unnecessary imports

* Code cleanup and update reference library link

* TPLink switch support (#2453)

* updated requirements

fabaff committed:
Upgrade googlemaps to 2.4.4 (#2481)

growers12 committed:
Implementation of a KNX platform driver and a KNX switch (#2439)

* Implementation of a KNX platform driver and a KNX switch

* Starting working on a KNX thermostat implementation

* Removed KNX thermostat implementation from this branch again

* Make gateway parameter optional (can be auto-detected in many cases)

* Removed check for double initialisation

* KNX messages now will be handled internally and not send to the Home Assistant message bus

* Call update_ha_state only if should_poll is false

* Removed unused HASS variable

* knxip library version changed

* pylint optimization

especially-relative committed:
Imap sensor (#2485)

* Imap unread email sensor

Checks the inbox of a imap account for unread emails. Tested against
gmail.

Example config:

```
sensor:
  - platform: imap
    name: gmail test
    user: USER
    password: PASSWORD
    server: imap.gmail.com
    port: 993

```

* added to .coveragerc

* Code cleanup and typo fix.

* Added port range validation

* Fix lint errors

Commit:
update rfxtrx to version 0.9 to support lighting4

Commit:
Make rfxtrx sensor not crash when unknown sensor is discovered

Danielhiversen committed:
Merge pull request #2487 from home-assistant/rfxtrx

Rfxtrx

balloob committed:
Tweak Recorder

turbokongen committed:
Missing command class for sensor (#2492)

balloob committed:
Update models.py

keatontaylor committed:
Clamp brightness between 0 and 255 (#2494)

* Clamp brightness between 0 and 255

Change to ensure that values over 255 supplied by the config will be
clamed to a max value of 255.

* Revert "Clamp brightness between 0 and 255"

This reverts commit c87238e8b5d3a67c035b1a97d5b5b9af8f6483bc.

* Clamp brightness between 0 and 255

Change to ensure that values over 255 supplied by the config will be
clamed to a max value of 255.

balloob committed:
Update frontend

balloob committed:
Update frontend

Danielhiversen committed:
Fix bug in rfxtrx for int device id (#2497)

turbokongen committed:
Add missing sensor command_class into sensor component (#2501)

command_class_sensor_alarm was also missing from sensor component.

fabaff committed:
Docstrings (#2502)

* Update docstrings

* Update docstrings

* Add link to docs

nkgilley committed:
Join by joaoapps component & notify platform (#2315)

* initial support for Join notifier

add more functions for Join

* rename to joaoapps_join

add message default in schema

move api_key check

* move special join services to their own component

update coveragerc and requirements_all

add icon and smallicon

fabaff committed:
Upgrade python-telegram-bot to 4.3.3 (#2504)

balloob committed:
Update frontend

rhooper committed:
Merge pull request #2489 from home-assistant/recorder-tests

Add more recorder tests

especially-relative committed:
add hvac mode support to radiotherm (#2442)

* add hvac mode support to radiotherm

off/cool/heat/auto modes are supported

* Moved STATE_AUTO to thermostat component, fix lint

Moved STATE_AUTO to thermostat platform. Fixed lint error.

Commit:
Allow device_tracker and sensor entity for google travel times (#2479)

* Allow owntracks entity for google travel times

* Added ability to use sensor state as location

* Added zone checks for google travel timesg

* Updated to use global constents and the location helper

* Fixed type in method name and removed redundant validation

* Changed domain condition to be a bit more elegant

* Updated to allow friendly name in any instance including the config

* Fixed bad python syntax and used helper methods

pvizeli committed:
add photo functionality to telegram (#2506)

* add photo functionality to telegram

* basic auth need password and username

robbiet480 committed:
Uber sensor now works with UberPool and has a bit cleaner logic. Also upgraded to latest version of the SDK and switched all single quotes to double quotes (#2507)

fabaff committed:
Upgrade zeroconf to 0.17.6 (#2503)

balloob committed:
Merge branch 'master' into dev

Conflicts:
	homeassistant/const.py

fabaff commented:
[Release blog post](https://github.com/home-assistant/home-assistant.io/blob/next/source/_posts/2016-07-15-tbd.markdown) synced...

fabaff committed:
Links docs (#2510)

* Add link to docs

* Fix link to docs

* Update docstrings

* Fix link

pavoni committed:
Bump pyloopenergy version.

fabaff committed:
Update docstrings (#2513)

Danielhiversen committed:
Fix issue #2290 for rfxtrx (#2498)

* Fix issue #2290 for rfxtrx

* update tests for rfxtrx sensor

* Replace state_unkown with None in rfxtrx sensor

* Update test_rfxtrx.py

balloob committed:
Update frontend (temp map solution)

pavoni committed:
Merge pull request #2514 from home-assistant/bump_pyloopenergy

Bump pyloopenergy version.

heytcass committed:
Update README.rst (#2517)

Editing for typos, clarifying.

turbokongen committed:
Zwave Node attributes was missing from binary sensors. (#2516)

Fixes #2505

turbokongen committed:
Fix node inclusion and exclusion. Also add secure inclusion. (#2519)

Fix node inclusion and exclusion.

pvizeli committed:
homematic update to pyhomematic 0.1.9

pvizeli committed:
Merge pull request #2520 from pvizeli/Homematic_pro

homematic update to pyhomematic 0.1.9

fabaff committed:
Switch to xmltodict and pass over missing temperature (fixes #2433) (#2463)

* Switch to xmltodict and pass over missing temperature (fixes #2433)

* Add guard clauses

balloob committed:
Update README.rst

kellerza committed:
Logbook: Query databse as_utc(). dt: Use pytz's localize (#2521)

balloob committed:
Update frontend

balloob committed:
update frontend

balloob committed:
Update frontend

balloob committed:
known devices yaml robustness (#2523)

michaelarnauts committed:
supported_media_commands should check for SERVICE_SELECT_SOURCE instead of SUPPORT_SELECT_SOURCE (#2482)

Commit:
Support for Wink lock user codes (#2525)

fabaff committed:
Catch ImportError (#2526)

pvizeli committed:
add media_player/clear_playlist and line-in/tv support to sonos (#2527)

* add media_player/clear_playlist and line-in/tv support to sonos

* add support source radio

* fix bug

* print TV/Line-In as media_title

* implement universal player

* add to demo platform

* Update demo.py

Better handling for demo object

* add unit tests

* fix unit test

fabaff committed:
Add validation and switch python-mystrom (#2529)

balloob committed:
Update frontend

fabaff committed:
Round output of wind speed and humidity (#2535)

pvizeli committed:
small bugfix (#2532)

kellerza committed:
Update rpi_gpio.py (#2530)

Should be pullup, since the sensor pulls to ground (at least the one on AndrewHilliday's site)

Or do we want this configurable?

balloob committed:
Make sqlalchemy main dependency to help migration (#2536)
